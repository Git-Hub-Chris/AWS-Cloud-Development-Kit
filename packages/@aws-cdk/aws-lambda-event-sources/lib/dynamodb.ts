import dynamodb = require('@aws-cdk/aws-dynamodb');
import lambda = require('@aws-cdk/aws-lambda');
import {StreamEventSource, StreamingEventSourceProps} from "./stream";

export interface DynamoEventSourceProps extends StreamingEventSourceProps {
}

/**
 * Use an Amazon DynamoDB stream as an event source for AWS Lambda.
 */
export class DynamoEventSource extends StreamEventSource {
  constructor(private readonly table: dynamodb.Table, props: DynamoEventSourceProps) {
    super(props, 1000);
  }

  public bind(target: lambda.IFunction) {
    if (!this.table.tableStreamArn) {
      throw new Error(`DynamoDB Streams must be enabled on the table ${this.table.node.path}`);
    }

    target.addEventSourceMapping(`DynamoDBEventSource:${this.table.node.uniqueId}`,
      this.getEventSourceMappingOptions(this.table.tableStreamArn)
    );

    this.table.grantStreamRead(target);
    dynamodb.Table.grantListStreams(target);
  }
}
