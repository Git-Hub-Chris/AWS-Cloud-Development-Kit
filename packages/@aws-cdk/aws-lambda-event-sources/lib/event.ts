import {Rule, RuleProps} from "@aws-cdk/aws-events";
import {LambdaFunction} from "@aws-cdk/aws-events-targets";
import {IEventSource, IFunction} from "@aws-cdk/aws-lambda";

import {Stack} from "@aws-cdk/core";

interface CloudWatchEventSourceProps extends RuleProps {
}

export class CloudWatchEventSource implements IEventSource {
    constructor(private props: CloudWatchEventSourceProps) {}

    /**
     * Binds a CloudWatch Events Rule with the given IFunction
     * @param target The IFunction to trigger with the event
     */
    public bind(target: IFunction): void {
        const stack = Stack.of(target);
        new Rule(stack, `${target.node.uniqueId}:ScheduleRule`, {
            ...this.props,
            targets: [...(this.props?.targets || []), new LambdaFunction(target)],
        });
    }
}
