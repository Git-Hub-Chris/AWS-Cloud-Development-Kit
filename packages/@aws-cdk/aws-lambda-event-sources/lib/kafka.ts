import * as crypto from 'crypto';
import { ISecurityGroup, IVpc, SubnetSelection } from '@aws-cdk/aws-ec2';
import * as iam from '@aws-cdk/aws-iam';
import * as lambda from '@aws-cdk/aws-lambda';
import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
import { StreamEventSource, StreamEventSourceProps } from './stream';

/**
 * Properties for a Kafka event source
 */
export interface KafkaEventSourceProps extends StreamEventSourceProps {
  /**
   * the Kafka topic to subscribe to
   */
  readonly topic: string,
  /**
   * the secret with the Kafka credentials, see https://docs.aws.amazon.com/msk/latest/developerguide/msk-password.html for details
   */
  readonly secret: secretsmanager.ISecret
}

/**
 * Properties for a MSK event source
 */
export interface ManagedKafkaEventSourceProps extends KafkaEventSourceProps {
  /**
   * the ARN of the MSK cluster
   */
  readonly clusterArn: string
}

/**
 * The authentication method to use with SelfManagedKafkaEventSource
 */
export enum AuthenticationMethod {
  /**
   * SASL_SCRAM_512_AUTH authentication method for your Kafka cluster
   */
  SASL_SCRAM_512_AUTH = 'SASL_SCRAM_512_AUTH',
  /**
   * SASL_SCRAM_256_AUTH authentication method for your Kafka cluster
   */
  SASL_SCRAM_256_AUTH = 'SASL_SCRAM_512_AUTH',
}

/**
 * Properties for a self managed Kafka cluster event source.
 * If your Kafka cluster is only reachable via VPC make sure to configure it.
 */
export interface SelfManagedKafkaEventSourceProps extends KafkaEventSourceProps {
  /**
   * The list of host and port pairs that are the addresses of the Kafka brokers in a "bootstrap" Kafka cluster that
   * a Kafka client connects to initially to bootstrap itself. They are in the format `abc.xyz.com:xxxx`.
   */
  readonly bootstrapServers: string[]

  /**
   * If your Kafka brokers are only reachable via VPC provide the VPC here
   *
   * @default none
   */
  readonly vpc?: IVpc;

  /**
   * If your Kafka brokers are only reachable via VPC, provide the subnets selection here
   *
   * @default - none, required if setting vpc
   */
  readonly vpcSubnets?: SubnetSelection,

  /**
   * If your Kafka brokers are only reachable via VPC, provide the security group here
   *
   * @default - none, required if setting vpc
   */
  readonly securityGroup?: ISecurityGroup

  /**
   * The authentication method for your Kafka cluster
   *
   * @default - SASL_SCRAM_512_AUTH
   */
  readonly authenticationMethod?: AuthenticationMethod
}

/**
 * Use a MSK cluster as a streaming source for AWS Lambda
 */
export class ManagedKafkaEventSource extends StreamEventSource {
  // This is to work around JSII inheritance problems
  private innerProps: ManagedKafkaEventSourceProps;

  constructor(props: ManagedKafkaEventSourceProps) {
    super(props);
    this.innerProps = props;
  }

  public bind(target: lambda.IFunction) {
    target.addEventSourceMapping(
      `KafkaEventSource:${this.innerProps.clusterArn}${this.innerProps.topic}`,
      this.enrichMappingOptions({
        eventSourceArn: this.innerProps.clusterArn,
        startingPosition: this.innerProps.startingPosition,
        sourceAccessConfigurations: [{ type: lambda.SourceAccessConfigurationType.SASL_SCRAM_512_AUTH, uri: this.innerProps.secret.secretArn }],
        kafkaTopic: this.innerProps.topic,
      }),
    );

    this.innerProps.secret.grantRead(target);

    target.addToRolePolicy(new iam.PolicyStatement(
      {
        actions: ['kafka:DescribeCluster', 'kafka:GetBootstrapBrokers', 'kafka:ListScramSecrets'],
        resources: [this.innerProps.clusterArn],
      },
    ));

    target.role?.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaMSKExecutionRole'));
  }
}

/**
 * Use a self hosted Kafka installation as a streaming source for AWS Lambda.
 */
export class SelfManagedKafkaEventSource extends StreamEventSource {
  // This is to work around JSII inheritance problems
  private innerProps: SelfManagedKafkaEventSourceProps;

  constructor(props: SelfManagedKafkaEventSourceProps) {
    super(props);
    if (props.vpc) {
      if (!props.securityGroup) {
        throw new Error('securityGroup must be set when providing vpc');
      }
      if (!props.vpcSubnets) {
        throw new Error('vpcSubnets must be set when providing vpc');
      }
    }
    this.innerProps = props;
  }

  public bind(target: lambda.IFunction) {
    let authenticationMethod;
    switch (this.innerProps.authenticationMethod) {
      case AuthenticationMethod.SASL_SCRAM_256_AUTH:
        authenticationMethod = lambda.SourceAccessConfigurationType.SASL_SCRAM_256_AUTH;
        break;
      case AuthenticationMethod.SASL_SCRAM_512_AUTH:
      default:
        authenticationMethod = lambda.SourceAccessConfigurationType.SASL_SCRAM_512_AUTH;
        break;
    }
    let sourceAccessConfigurations = [{ type: authenticationMethod, uri: this.innerProps.secret.secretArn }];
    if (this.innerProps.vpcSubnets !== undefined && this.innerProps.securityGroup !== undefined) {
      sourceAccessConfigurations.push({
        type: lambda.SourceAccessConfigurationType.VPC_SECURITY_GROUP,
        uri: this.innerProps.securityGroup.securityGroupId,
      },
      );
      this.innerProps.vpc?.selectSubnets(this.innerProps.vpcSubnets).subnetIds.forEach((id) => {
        sourceAccessConfigurations.push({ type: lambda.SourceAccessConfigurationType.VPC_SUBNET, uri: id });
      });
    }
    const idHash = crypto.createHash('md5').update(JSON.stringify(this.innerProps.bootstrapServers)).digest('hex');
    target.addEventSourceMapping(
      `KafkaEventSource:${idHash}:${this.innerProps.topic}`,
      this.enrichMappingOptions({
        kafkaBootstrapServers: this.innerProps.bootstrapServers,
        kafkaTopic: this.innerProps.topic,
        startingPosition: this.innerProps.startingPosition,
        sourceAccessConfigurations,
      }),
    );
    this.innerProps.secret.grantRead(target);
  }
}
