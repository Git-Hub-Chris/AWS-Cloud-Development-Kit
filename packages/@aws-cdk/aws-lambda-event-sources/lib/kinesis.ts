import * as kinesis from '@aws-cdk/aws-kinesis';
import * as lambda from '@aws-cdk/aws-lambda';
import * as cdk from '@aws-cdk/core';
import { StreamEventSource, StreamEventSourceProps } from './stream';

export interface KinesisEventSourceProps extends StreamEventSourceProps {
  /**
   * A list of event filtering criteria to control which events Lambda sends to
   * your function for processing. You can define up to five different filters
   * for a single event source. If an event satisfies any one of these five
   * filters, Lambda sends the event to your function. Otherwise, Lambda
   * discards the event. An event either satisfies the filter criteria or it
   * doesn't. If you're using batching windows, Lambda applies your filter
   * criteria to each new event to determine whether to add it to the current
   * batch.
   *
   * @see https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html
   *
   * @default - none
   *
   * @example
   * import * as kinesis from '@aws-cdk/aws-kinesis';
   * import * as sources from '@aws-cdk/aws-lambda-event-sources';
   * declare const stream: kinesis.IStream;
   * new sources.KinesisEventSource(stream, {
   *   startingPosition: lambda.StartingPosition.TRIM_HORIZON,
   *   filterPatterns: [
   *     { data: { balance: [{ numeric: [">", 500] }] } }
   *   ]
   * });
   */
  readonly filterPatterns?: lambda.FilterPattern[];
}

/**
 * Use an Amazon Kinesis stream as an event source for AWS Lambda.
 */
export class KinesisEventSource extends StreamEventSource {
  private _eventSourceMappingId?: string = undefined;
  private innerProps: KinesisEventSourceProps;

  constructor(readonly stream: kinesis.IStream, props: KinesisEventSourceProps) {
    super(props);

    this.props.batchSize !== undefined && cdk.withResolved(this.props.batchSize, batchSize => {
      if (batchSize < 1 || batchSize > 10000) {
        throw new Error(`Maximum batch size must be between 1 and 10000 inclusive (given ${this.props.batchSize})`);
      }
    });
    if ((props.filterPatterns?.length ?? 0) > 5) {
      throw new Error(`Maximum number of filter criteria for a single event source is five (given ${props.filterPatterns?.length}).`);
    }
    this.innerProps = props;
  }

  protected enrichMappingOptions(options: lambda.EventSourceMappingOptions): lambda.EventSourceMappingOptions {
    const eventSourceMapping = super.enrichMappingOptions(options);
    return {
      ...eventSourceMapping,
      filterPatterns: this.innerProps.filterPatterns,
    };
  }

  public bind(target: lambda.IFunction) {
    const eventSourceMapping = target.addEventSourceMapping(`KinesisEventSource:${cdk.Names.nodeUniqueId(this.stream.node)}`,
      this.enrichMappingOptions({ eventSourceArn: this.stream.streamArn }),
    );
    this._eventSourceMappingId = eventSourceMapping.eventSourceMappingId;

    this.stream.grantRead(target);

    // The `grantRead` API provides all the permissions recommended by the Kinesis team for reading a stream.
    // `DescribeStream` permissions are not required to read a stream as it's covered by the `DescribeStreamSummary`
    // and `SubscribeToShard` APIs.
    // The Lambda::EventSourceMapping resource validates against the `DescribeStream` permission. So we add it explicitly.
    // FIXME This permission can be removed when the event source mapping resource drops it from validation.
    this.stream.grant(target, 'kinesis:DescribeStream');
  }

  /**
   * The identifier for this EventSourceMapping
   */
  public get eventSourceMappingId(): string {
    if (!this._eventSourceMappingId) {
      throw new Error('KinesisEventSource is not yet bound to an event source mapping');
    }
    return this._eventSourceMappingId;
  }
}
