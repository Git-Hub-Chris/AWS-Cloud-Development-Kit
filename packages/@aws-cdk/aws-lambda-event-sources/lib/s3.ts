import lambda = require('@aws-cdk/aws-lambda');
import s3 = require('@aws-cdk/aws-s3');

export interface S3EventSourceProps {
  events: s3.EventType[];
  filters?: s3.NotificationKeyFilter[];
}

export class S3EventSource implements lambda.IEventSource {
  constructor(readonly bucket: s3.Bucket, private readonly props: S3EventSourceProps) {

  }

  public bind(target: lambda.FunctionRef) {
    const filters = this.props.filters || [];
    for (const event of this.props.events) {
      this.bucket.onEvent(event, target, ...filters);
    }
  }
}
