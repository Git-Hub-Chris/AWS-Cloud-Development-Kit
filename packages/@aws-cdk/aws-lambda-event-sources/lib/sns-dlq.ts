import { DlqDestinationConfig, IEventSourceDlq } from "@aws-cdk/aws-lambda";
import * as sns from '@aws-cdk/aws-sns';

/**
 * An SNS dead letter queue destination configuration for a Lambda event source
 */
export class SnsDlq implements IEventSourceDlq {
  constructor(private readonly topic: sns.ITopic) {
  }

  /**
   * Returns a destination configuration for the DLQ
   */
  public bind(): DlqDestinationConfig {
    return {
      destination: this.topic.topicArn
    };
  }
}
