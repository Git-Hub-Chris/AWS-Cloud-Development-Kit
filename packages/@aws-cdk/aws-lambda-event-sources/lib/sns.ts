import lambda = require('@aws-cdk/aws-lambda');
import sns = require('@aws-cdk/aws-sns');

/**
 * An SNS topic as an event source for AWS Lambda function.
 *
 * To use:
 *
 *    lambda.addEventSource(new SnsEventSource(topic));
 *
 */
export class SnsEventSource implements lambda.IEventSource {
  constructor(readonly topic: sns.TopicRef) {
  }

  public bind(target: lambda.FunctionRef) {
    this.topic.subscribeLambda(target);
  }
}