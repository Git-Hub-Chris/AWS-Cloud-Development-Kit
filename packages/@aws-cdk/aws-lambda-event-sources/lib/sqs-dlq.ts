import { DlqDestinationConfig, IEventSourceDlq } from "@aws-cdk/aws-lambda";
import * as sqs from '@aws-cdk/aws-sqs';

/**
 * An SQS dead letter queue destination configuration for a Lambda event source
 */
export class SqsDlq implements IEventSourceDlq {
    constructor(private readonly queue: sqs.IQueue) {
    }

    /**
     * Returns a destination configuration for the DLQ
     */
    public bind(): DlqDestinationConfig {
        return {
            destination: this.queue.queueArn
        };
    }
}
