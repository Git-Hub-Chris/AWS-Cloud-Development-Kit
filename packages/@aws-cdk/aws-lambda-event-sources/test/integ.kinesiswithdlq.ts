import * as kinesis from '@aws-cdk/aws-kinesis';
import * as lambda from '@aws-cdk/aws-lambda';
import * as sqs from '@aws-cdk/aws-sqs';
import { App, Stack } from "@aws-cdk/core";
import * as path from 'path';
import { KinesisEventSource, SqsDlq } from '../lib';

/*
 * Stack verification steps:
 * * aws lambda invoke --function-name <deployed 'FP' function name> --invocation-type Event --payload '"Event"' response.json
 * * Validate two executions of fn lambda
 * * Validate one message in dlq
 */

// tslint:disable:no-console
async function handler(event: any) {
  console.log('event:', JSON.stringify(event, undefined, 2));
  throw new Error();
}

class KinesisWithDLQTest extends Stack {
  constructor(scope: App, id: string) {
    super(scope, id);

    const fn = new lambda.Function(this, 'F', {
      runtime: lambda.Runtime.NODEJS_10_X,
      handler: 'index.handler',
      code: lambda.Code.fromInline(`exports.handler = ${handler.toString()}`)
    });

    const stream = new kinesis.Stream(this, 'S');

    const dlq = new sqs.Queue(this, 'Q');

    fn.addEventSource(new KinesisEventSource(stream, {
      startingPosition: lambda.StartingPosition.TRIM_HORIZON,
      onFailure: new SqsDlq(dlq),
      retryAttempts: 2
    }));

    new lambda.Function(this, 'FP', {
      runtime: lambda.Runtime.NODEJS_10_X,
      handler: 'index.handler',
      code: lambda.AssetCode.fromAsset(path.join(__dirname, 'integ.kinesiswithdlq.handler'))
    });
  }
}

const app = new App();
new KinesisWithDLQTest(app, 'lambda-event-source-kinesis-with-dlq');
app.synth();
