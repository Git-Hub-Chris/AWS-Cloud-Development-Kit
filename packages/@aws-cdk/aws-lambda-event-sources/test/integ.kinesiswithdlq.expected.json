{
  "Resources": {
    "FServiceRole3AC82EE1": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
              ]
            ]
          }
        ]
      }
    },
    "FServiceRoleDefaultPolicy17A19BFA": {
      "Type": "AWS::IAM::Policy",
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "sqs:SendMessage",
                "sqs:GetQueueAttributes",
                "sqs:GetQueueUrl"
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "Q63C6E3AB",
                  "Arn"
                ]
              }
            },
            {
              "Action": [
                "kinesis:DescribeStream",
                "kinesis:GetRecords",
                "kinesis:GetShardIterator"
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::GetAtt": [
                  "S509448A1",
                  "Arn"
                ]
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "PolicyName": "FServiceRoleDefaultPolicy17A19BFA",
        "Roles": [
          {
            "Ref": "FServiceRole3AC82EE1"
          }
        ]
      }
    },
    "FC4345940": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "ZipFile": "exports.handler = async function handler(event) {\n    // tslint:disable-next-line:no-console\n    console.log('event:', JSON.stringify(event, undefined, 2));\n    throw new Error();\n}"
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "FServiceRole3AC82EE1",
            "Arn"
          ]
        },
        "Runtime": "nodejs10.x"
      },
      "DependsOn": [
        "FServiceRoleDefaultPolicy17A19BFA",
        "FServiceRole3AC82EE1"
      ]
    },
    "FKinesisEventSourcelambdaeventsourcekinesiswithdlqSD357FCB87EEA8CB4": {
      "Type": "AWS::Lambda::EventSourceMapping",
      "Properties": {
        "EventSourceArn": {
          "Fn::GetAtt": [
            "S509448A1",
            "Arn"
          ]
        },
        "FunctionName": {
          "Ref": "FC4345940"
        },
        "BatchSize": 100,
        "DestinationConfig": {
          "OnFailure": {
            "Destination": {
              "Fn::GetAtt": [
                "Q63C6E3AB",
                "Arn"
              ]
            }
          }
        },
        "MaximumRetryAttempts": 0,
        "StartingPosition": "TRIM_HORIZON"
      }
    },
    "S509448A1": {
      "Type": "AWS::Kinesis::Stream",
      "Properties": {
        "ShardCount": 1,
        "RetentionPeriodHours": 24
      }
    },
    "Q63C6E3AB": {
      "Type": "AWS::SQS::Queue"
    },
    "FPServiceRoleB1957CC2": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
              ]
            ]
          }
        ]
      }
    },
    "FP85AE6139": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Ref": "AssetParameters013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604S3BucketF010E518"
          },
          "S3Key": {
            "Fn::Join": [
              "",
              [
                {
                  "Fn::Select": [
                    0,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParameters013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604S3VersionKeyB38C926C"
                        }
                      ]
                    }
                  ]
                },
                {
                  "Fn::Select": [
                    1,
                    {
                      "Fn::Split": [
                        "||",
                        {
                          "Ref": "AssetParameters013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604S3VersionKeyB38C926C"
                        }
                      ]
                    }
                  ]
                }
              ]
            ]
          }
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "FPServiceRoleB1957CC2",
            "Arn"
          ]
        },
        "Runtime": "nodejs10.x"
      },
      "DependsOn": [
        "FPServiceRoleB1957CC2"
      ]
    }
  },
  "Outputs": {
    "InputKinesisStreamName": {
      "Value": {
        "Ref": "S509448A1"
      }
    },
    "DlqSqsQueueUrl": {
      "Value": {
        "Ref": "Q63C6E3AB"
      }
    }
  },
  "Parameters": {
    "AssetParameters013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604S3BucketF010E518": {
      "Type": "String",
      "Description": "S3 bucket for asset \"013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604\""
    },
    "AssetParameters013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604S3VersionKeyB38C926C": {
      "Type": "String",
      "Description": "S3 key for asset version \"013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604\""
    },
    "AssetParameters013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604ArtifactHash5300B2A6": {
      "Type": "String",
      "Description": "Artifact hash for asset \"013549436de3944a6176501ce5de71aef355a25e669d5af45fd69f29785df604\""
    }
  }
}